benchmarks/time_solve_models.py

```diff
@@ -22,11 +22,328 @@ def prepare_model(model):
+    geometry = model.default_geometry
+
+    # load parameter values and process model and geometry
+    param = pybamm.ParameterValues("Ramadass2004")
+    param.process_model(model)
+    param.process_geometry(geometry)
+
+    # set mesh
+    var_pts = {"x_n": 20, "x_s": 20, "x_p": 20, "r_n": 30, "r_p": 30, "y": 10, "z": 10}
+    mesh = pybamm.Mesh(geometry, model.default_submesh_types, var_pts)
+
+    # discretise model
+    disc = pybamm.Discretisation(mesh, model.default_spatial_methods)
+    disc.process_model(model)
+
+
+def prepare_model_Mohtat2020(model):
+    geometry = model.default_geometry
+
+    # load parameter values and process model and geometry
+    param = pybamm.ParameterValues("Mohtat2020")
+    param.process_model(model)
+    param.process_geometry(geometry)
+
+    # set mesh
+    var_pts = {"x_n": 20, "x_s": 20, "x_p": 20, "r_n": 30, "r_p": 30, "y": 10, "z": 10}
+    mesh = pybamm.Mesh(geometry, model.default_submesh_types, var_pts)
+
+    # discretise model
+    disc = pybamm.Discretisation(mesh, model.default_spatial_methods)
+    disc.process_model(model)
+
+
+def prepare_model_Chen2020(model):
+    geometry = model.default_geometry
+
+    # load parameter values and process model and geometry
+    param = pybamm.ParameterValues("Chen2020")
+    param.process_model(model)
+    param.process_geometry(geometry)
+
+    # set mesh
+    var_pts = {"x_n": 20, "x_s": 20, "x_p": 20, "r_n": 30, "r_p": 30, "y": 10, "z": 10}
+    mesh = pybamm.Mesh(geometry, model.default_submesh_types, var_pts)
+
+    # discretise model
+    disc = pybamm.Discretisation(mesh, model.default_spatial_methods)
+    disc.process_model(model)
+
+
+def prepare_model_Chen2020_plating(model):
+    geometry = model.default_geometry
+
+    # load parameter values and process model and geometry
+    param = pybamm.ParameterValues("Chen2020_plating")
+    param.process_model(model)
+    param.process_geometry(geometry)
+
+    # set mesh
+    var_pts = {"x_n": 20, "x_s": 20, "x_p": 20, "r_n": 30, "r_p": 30, "y": 10, "z": 10}
+    mesh = pybamm.Mesh(geometry, model.default_submesh_types, var_pts)
+
+    # discretise model
+    disc = pybamm.Discretisation(mesh, model.default_spatial_methods)
+    disc.process_model(model)
+
+
+def prepare_model_Ecker2015(model):
+    geometry = model.default_geometry
+
+    # load parameter values and process model and geometry
+    param = pybamm.ParameterValues("Ecker2015")
+    param.process_model(model)
+    param.process_geometry(geometry)
+
+    # set mesh
+    var_pts = {"x_n": 20, "x_s": 20, "x_p": 20, "r_n": 30, "r_p": 30, "y": 10, "z": 10}
+    mesh = pybamm.Mesh(geometry, model.default_submesh_types, var_pts)
+
+    # discretise model
+    disc = pybamm.Discretisation(mesh, model.default_spatial_methods)
+    disc.process_model(model)
+
+
 def solve_model_once(model, solver, t_eval):
     solver.solve(model, t_eval=t_eval)
 
 
-class TimeSolveSPM:
+class TimeSolveSPMMarquis2019:
```

Comment:
Same as before, but need to be careful as you will have two `parameter` sets for the tests: the battery parameters and the solve (True, False). In the [docs](https://asv.readthedocs.io/en/stable/writing_benchmarks.html#parameterized-benchmarks) there is an example on how to do it.